packages/babel-plugin-transform-classes/src/vanilla.js

```diff
@@ -152,14 +153,21 @@ export default class ClassTransformer {
       if (body.length === 1) return t.toExpression(body[0]);
     }
 
-    //
-    body.push(t.returnStatement(this.classRef));
+    const iife = getClassIIFE(this.path.node);
 
-    const container = t.arrowFunctionExpression(
-      closureParams,
-      t.blockStatement(body),
-    );
-    return t.callExpression(container, closureArgs);
+    iife.arguments.push(...closureArgs);
+    iife.callee.params.push(...closureParams);
+
+    if (iife.callee.body.body.length === 0) {
+      iife.callee.body.body.push(...body, t.returnStatement(this.classRef));
+      return iife;
+    }
+
+    if (this.path.isClassExpression()) {
```

Comment:
hating this `IfStatement` here, kinda a hack to insert a correct return for mentioned double IIFEs case - need to figure out how to flatten those IIFEs (in case of `ClassExpresion` with static properties) and keep all surrounding references correct

Reply:
https://github.com/babel/babel/blob/master/packages/babel-traverse/src/scope/lib/renamer.js#L42-L85 looks similar